packages/core/src/components/AutocompleteField/AutocompleteField.jsx

```diff
@@ -0,0 +1,175 @@
+import Button from '../Button/Button';
+import Downshift from 'downshift';
+import FormLabel from '../FormLabel/FormLabel';
+import PropTypes from 'prop-types';
+import React from 'react';
+import noop from 'nooop';
+import uniqueId from 'lodash.uniqueid';
+
+/**
+ * An `AutocompleteField` component renders a typeahead input field, form label, and a clear input link.
+ */
+export class AutocompleteField extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.id = uniqueId('autocomplete_');
+  }
+
+  render() {
+    const {
+      clearAriaLabel,
+      clearInputText,
+      disabled,
+      fieldRef,
+      items,
+      itemToString,
+      labelHint,
+      labelText,
+      onChange,
+      onStateChange,
+      ...fieldProps
+    } = this.props;
+
+    return (
+      <Downshift
+        itemToString={itemToString}
+        onChange={onChange}
+        onStateChange={onStateChange}
+        render={({
+          clearSelection,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          isOpen,
+          inputValue
+        }) => (
+          <div className="ds-u-display--flex ds-u-flex-wrap--wrap ds-c-autocomplete">
+            <FormLabel
+              className="ds-u-margin-top--0"
+              labelClassName="ds-c-autocomplete__label"
+              component="label"
+              fieldId={this.id}
+              hint={labelHint}
+            >
+              {labelText}
+            </FormLabel>
+
+            <input
+              {...getInputProps({
+                className: 'ds-c-field ds-c-autocomplete__input',
+                disabled: disabled,
+                id: this.id,
+                ref: fieldRef,
+                ...fieldProps
+              })}
+            />
+
```

Comment:
Would it be better if this component accepted a `TextField` component as a child instead? If it's possible, I personally favor reusing an existing component, and then we can remove all the label and input props from `Autocomplete`, and someone could potentially later add the autocomplete functionality to an existing form field by wrapping it with the `Autocomplete` component:

```jsx
<Autocomplete items=["A", "B", "C"]>
  <TextField defaultValue="Foo" label ="Enter a zipcode" hint="For example: 12345" />
</Autocomplete>
```

You can check out `Tabs.jsx` to see an example of how you might manipulate the `children` prop.

Reply:
@pwolfert pointed me to Downshift source line 649 ( https://github.com/paypal/downshift/blob/master/src/downshift.js#L649 ) and 612, where `getInputProps` is defined. This method applies the key role, aria, and autocomplete attributes, and is what makes Downshift fully accessible. 

We need access to Downshift's `getInputProps` and those can’t be directly applied to `TextField` children unless we add a generic `inputProps` prop to `TextField` and pass them down to the `<input />`.

Reply:
This isn't documented anywhere (right now), but `TextField` already accepts any prop and they get passed into the `input` field: https://github.com/CMSgov/design-system/blob/d556a674939f892c8c83e77c3c822ff321b4ee07/packages/core/src/components/TextField/TextField.jsx#L68

```jsx
const textFieldChild = this.props.children; // we'd want to filter this to get <TextField>
const inputProps = getInputProps();
const field = React.cloneElement(textFieldChild, {...inputProps});
```

Reply:
I'll give it a try, and see what happens.